### PR TITLE
SWARM-1878 - Update OpenTracing and Jaeger versions

### PR DESCRIPTION
**Pull request by jpkrohling**
Signed-off-by: Juraci Paixão Kröhling <juraci@kroehling.de>

- [x] Have you followed the guidelines in our [Contributing](http://wildfly-swarm.io/community/contributing/) document?
- [x] Have you created a [JIRA](https://issues.jboss.org/browse/SWARM) and used it in the commit message?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/wildfly-swarm/wildfly-swarm/pulls) for the same issue?
- [x] Have you built the project locally prior to submission with `mvn clean install`?

-----

~~This is currently WIP, pending release of Jaeger Client Java (jaegertracing/jaeger-client-java#365))~~

### CODE REVIEW

File: build-parent/pom.xml [hunk 1]
```diff
@@ -82,10 +82,9 @@
     <version.zipkin>1.13.1</version.zipkin>
 
     <!-- OpenTracing related versions -->
-    <version.opentracing.tracerresolver>0.1.0</version.opentracing.tracerresolver>
-    <version.opentracing.servlet>0.0.9</version.opentracing.servlet>
-    <version.opentracing.spanmanager>0.0.5</version.opentracing.spanmanager>
-    <version.jaeger>0.20.0</version.jaeger>
+    <version.opentracing.tracerresolver>0.1.4</version.opentracing.tracerresolver>
+    <version.opentracing.servlet>0.1.0</version.opentracing.servlet>
+    <version.jaeger>0.24.1-SNAPSHOT</version.jaeger>
```
Comment (kenfinnigan) [inline build-parent/pom.xml]: Did you mean to use a SNAPSHOT version?

We can't release with a SNAPSHOT

**Comment by jpkrohling** [comment]
@objectiser , could you please review the changes? I'll amend the commit once Jaeger Client Java 0.25.0 is released, but I wanted to check if the other changes look sane to you.

File: build-parent/pom.xml (hunk 1 above)
Comment (jpkrohling) [reply build-parent/pom.xml]: > This is currently WIP, pending release of Jaeger Client Java (jaegertracing/jaeger-client-java#365))

:wink:

File: testsuite/testsuite-opentracing/src/test/java/org/wildfly/swarm/opentracing/MockTracerResolver.java [hunk 2]
```diff
@@ -3,13 +3,13 @@
...
 import io.opentracing.mock.MockTracer;
-import io.opentracing.util.ThreadLocalActiveSpanSource;
+import io.opentracing.util.ThreadLocalScopeManager;
 
 /**
  * @author Juraci Paixão Kröhling
  */
 public class MockTracerResolver extends TracerResolver {
-    static final MockTracer TRACER_INSTANCE = new MockTracer(new ThreadLocalActiveSpanSource(), MockTracer.Propagator.TEXT_MAP);
+    static final MockTracer TRACER_INSTANCE = new MockTracer(new ThreadLocalScopeManager(), MockTracer.Propagator.TEXT_MAP);
```
Comment (objectiser) [inline testsuite/testsuite-opentracing/src/test/java/org/wildfly/swarm/opentracing/MockTracerResolver.java]: Can use the default constructor now, which uses these same parameters.

**Review by objectiser — COMMENTED** [review]
Other than the one comment, the changes look good.

File: testsuite/testsuite-opentracing/src/test/java/org/wildfly/swarm/opentracing/MockTracerResolver.java (hunk 2 above)
Comment (jpkrohling) [reply testsuite/testsuite-opentracing/src/test/java/org/wildfly/swarm/opentracing/MockTracerResolver.java]: Thanks!

**Comment by jpkrohling** [comment]
@kenfinnigan , after the rebase, it seems that I can't run `mvn clean install` anymore:

```
Failed pattern wfswarm\S+[0-9]{5,}.\S{5,} in directory /tmp./tmp/wfswarmwildfly-self-contained820108085243930922.d size(80) too large. Max. is 0
Failed pattern wfswarm\S+[0-9]{5,}.\S{5,} in directory /tmp./tmp/wfswarmmodule-jar5498856170619631873.jar_d size(100) too large. Max. is 0
Failed pattern wfswarm\S+[0-9]{5,}.\S{5,} in directory /tmp./tmp/wfswarmmodule-jar2832827519316906182.jar_d size(160) too large. Max. is 0
Failed pattern wfswarm\S+[0-9]{5,}.\S{5,} in directory /tmp./tmp/wfswarmwildfly-self-contained5757732020001915857.d size(80) too large. Max. is 0
Failed pattern wfswarm\S+[0-9]{5,}.\S{5,} in directory /tmp./tmp/wfswarmmodule-jar2838691799278949942.jar_d size(100) too large. Max. is 0
Failed pattern wfswarm\S+[0-9]{5,}.\S{5,} in directory /tmp./tmp/wfswarmmodule-jar2033138414276140994.jar_d size(160) too large. Max. is 0
```

Is this expected? In any case, it's now ready to be reviewed by the Swarm team and eventually merged.

**Comment by juangon** [comment]
@jpkrohling this was caused by a merge yesterday it seems. I am fixing it. Which test is failing?

**Comment by jpkrohling** [comment]
It fails right at the beginning. Here's a complete output for `mvn clean install` from my master:

https://paste.fedoraproject.org/paste/bwmJvCMvAyxi8OZj7X01bw

**Comment by juangon** [comment]
@jpkrohling can you try removing all files/directories starting with wfswarm in your temp directory? (/tmp) and retry?

**Comment by jpkrohling** [comment]
@juangon it seems to have worked! And it looks like it passed on Jenkins/Linux as well.

**Comment by juangon** [comment]
great @jpkrohling Windows failure is fixed by https://github.com/wildfly-swarm/wildfly-swarm/pull/830 so no problem about it
/cc @kenfinnigan

**Review by kenfinnigan — APPROVED** [review]

**Pull request merged**